### DISCUSSION TITLE
How can I show full path of the current file in Status Line?

### DISCUSSION THREAD

**Discussion by nhanluongoe** [NvChad/NvChad#2246] [Q&A] [answered]
For example in this picture. It's only showing `default_config.lua`, how can I show the full path of the file like `a/b/default_config.lua`?

<img width="392" alt="image" src="https://github.com/NvChad/NvChad/assets/42910096/8328ed8f-98ff-462e-98ea-1112c544bcaf">

I appreciate your help.

**Comment by georgejean** [comment, marked as answer]
Run `NvChadUpdate`. Then you can add this in your `M.ui` of your `chadrc.lua` file:
```lua
  statusline = {
    overriden_modules = function(modules)
      modules[2] = (function()
        local fn = vim.fn
        local config = require("core.utils").load_config().ui.statusline
        local sep_style = config.separator_style

        local default_sep_icons = {
          default = { left = "", right = " " },
          round = { left = "", right = "" },
          block = { left = "█", right = "█" },
          arrow = { left = "", right = "" },
        }

        local separators = (type(sep_style) == "table" and sep_style) or default_sep_icons[sep_style]

        local sep_r = separators["right"]
        local icon = " 󰈚 "
        local filename = (fn.expand "%" == "" and "Empty ") or fn.expand "%:p" --change here

        if filename ~= "Empty " then
          local devicons_present, devicons = pcall(require, "nvim-web-devicons")

          if devicons_present then
            local ft_icon = devicons.get_icon(filename)
            icon = (ft_icon ~= nil and " " .. ft_icon) or ""
          end

          filename = " " .. filename .. " "
        end

        return "%#St_file_info#" .. icon .. filename .. "%#St_file_sep#" .. sep_r
      end)()
    end,
  },
```
I've just replaced the "%:t" [here](https://github.com/NvChad/ui/blob/1d4267f47fc022c66af87261f49c32b9a05273cf/lua/nvchad_ui/statusline/default.lua#L73) by "%:p".
See also the [doc](https://nvchad.com/docs/config/nvchad_ui) and [here](https://github.com/NvChad/ui/blob/1d4267f47fc022c66af87261f49c32b9a05273cf/lua/nvchad_ui/statusline/default.lua#L186C3-L199C4) for the statusline modules' indices (here, I've edited the module 2 `fileInfo`).

**Reply by cfcy** [reply]
How to show the relative path, ie, the path in nvim-tree? Thank you.

**Reply by georgejean** [reply]
Hi ! For the relative path, you can simply replace `%:p` by `%:.` in the code above. See `:help %:`
